src/roiextractors/extractors/schnitzerextractor/extractsegmentationextractor.py

```diff
@@ -7,22 +14,216 @@
 except ImportError:
     HAVE_H5PY = False
 
-import numpy as np
-from lazy_ops import DatasetView
 
-from ...extraction_tools import PathType
+from ...extraction_tools import PathType, ArrayType
 from ...multisegmentationextractor import MultiSegmentationExtractor
 from ...segmentationextractor import SegmentationExtractor
 
 
-class ExtractSegmentationExtractor(SegmentationExtractor):
+class ExtractSegmentationExtractor(ABC):
+    """Abstract class that defines which extractor class to use for a given file."""
+
+    extractor_name = "ExtractSegmentation"
+    installed = HAVE_H5PY  # check at class level if installed or not
+    installation_mesg = "To use ExtractSegmentationExtractor install h5py: \n\n pip install h5py \n\n"  # error message when not installed
+
+    def __new__(
+        cls,
+        file_path: PathType,
+        sampling_frequency: float,
```

Comment:
@bendichter Is making this a required attribute a good choice when we have irregular timing?
Another example based on [EXTRACT_output_to_nwb.m]:
(https://github.com/schnitzer-lab/EXTRACT-public/blob/master/EXTRACT/helper_functions/EXTRACT_output_to_nwb.m)
https://github.com/schnitzer-lab/EXTRACT-public/blob/d7a83736918fbef24643860ec6ac3aabfb71c617/EXTRACT/helper_functions/EXTRACT_output_to_nwb.m#L46-L52
maybe we could also have `sampling_frequency` and `timestamps` and then add a check within the class that either of them must be supplied. Also correct me if I'm wrong about this, but I don't think currently using timestamps with a `SegmentationExtractor` is supported. Do you think it will make sense to have something similar as it is done in `add_two_photon_series`? 
https://github.com/catalystneuro/neuroconv/blob/69fff3b45828dd27df65cdeea92ab062140dc283/src/neuroconv/tools/roiextractors/roiextractors.py#L418-L425

Reply:
> maybe we could also have sampling_frequency and timestamps and then add a check within the class that either of them must be supplied

that's a good idea

Reply:
> Do you think it will make sense to have something similar as it is done in add_two_photon_series?

Yes, that might be necessary